current.go

```diff
@@ -61,32 +52,111 @@ func (r *VaultRole) Search(config SearchConfig) (Role, error) {
 		return Role{}, microerror.Maskf(notFoundError, "no vault secret at path '%s'", key.RoleName(config.ID, config.Organizations))
 	}
 
-	b, err := json.Marshal(secret.Data)
+	role, err := vaultSecretToRole(secret)
 	if err != nil {
 		return Role{}, microerror.Mask(err)
 	}
 
-	var internalRole role
-	err = json.Unmarshal(b, &internalRole)
-	if err != nil {
-		return Role{}, microerror.Mask(err)
+	role.ID = config.ID
+	return role, nil
+}
+
+// vaultSecretToRole makes required type casts / type checks and parsing to
+// extract role information from Vault api.Secret.
+func vaultSecretToRole(secret *api.Secret) (Role, error) {
+	var role Role
+
+	{
+		v, exists := secret.Data["allow_bare_domains"]
+		if !exists {
+			return Role{}, microerror.Maskf(invalidConfigError, "allow_bare_domains missing from Vault api.Secret.Data")
```

Comment:
So far the `invalidConfigError` was used for program initialization. For arbitrary runtime errors we just do want to annotate we use `executionFailedError`. To be more specific we could have a `vaultResponseFormatError` or something like that.

Reply:
Gosh, only noticed this from email after reading PR comments and merged. Will fix this too.